libs/common/src/lib/tasks-automation/task-details/task-testing/task-testing.component.ts

```diff
@@ -0,0 +1,146 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, inject, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {
+  InfoCardComponent,
+  SisProgressModule,
+  SisToastService,
+  TwoColumnsConfigurationItemComponent,
+} from '@nuclia/sistema';
+import { SDKService } from '@flaps/core';
+import { TranslateModule } from '@ngx-translate/core';
+import { TaskWithApplyOption } from '../../task-route.directive';
+import {
+  PaButtonModule,
+  PaDropdownModule,
+  PaPopupModule,
+  PaTableModule,
+  PaTextFieldModule,
+  PaTogglesModule,
+} from '@guillotinaweb/pastanaga-angular';
+import { AugmentedField, IResource, Resource } from '@nuclia/core';
+import { debounceTime, forkJoin, map, of, Subject, switchMap, take } from 'rxjs';
+import { TestResultsComponent } from './test-results/test-results.component';
+import { getTaskFilterTypeFromTaskName } from '../../tasks-automation.models';
+
+export type TestResults = { resource: Resource; results: { [key: string]: AugmentedField } };
+
+@Component({
+  selector: 'app-task-testing',
+  imports: [
+    CommonModule,
+    InfoCardComponent,
+    PaButtonModule,
+    PaDropdownModule,
+    PaPopupModule,
+    PaTableModule,
+    PaTextFieldModule,
+    PaTogglesModule,
+    SisProgressModule,
+    TestResultsComponent,
+    TwoColumnsConfigurationItemComponent,
+    TranslateModule,
+  ],
+  templateUrl: './task-testing.component.html',
+  styleUrl: './task-testing.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TaskTestingComponent {
+  private sdk = inject(SDKService);
+  private cdr = inject(ChangeDetectorRef);
+  private toaster = inject(SisToastService);
+
+  updateSuggestions = new Subject<string>();
+  resources: IResource[] = [];
+  selected: { [id: string]: boolean } = {};
+  loading = false;
+  results?: TestResults[];
+  tokens?: { input: number; output: number };
+
+  @Input() task: TaskWithApplyOption | undefined;
+
+  suggestions = this.updateSuggestions.pipe(
+    debounceTime(300),
+    switchMap((value) =>
+      this.sdk.currentKb.pipe(
+        take(1),
+        switchMap((kb) =>
+          value.length > 2
+            ? kb.catalog(value).pipe(map((res) => (res.type === 'error' ? [] : Object.values(res.resources || {}))))
+            : of([]),
+        ),
+      ),
+    ),
+  );
+
+  get selectedIds() {
+    return Object.entries(this.selected)
+      .filter(([, enabled]) => enabled)
+      .map(([key]) => key);
+  }
+
+  run() {
+    this.loading = true;
+    this.results = undefined;
+    this.tokens = undefined;
+    this.cdr.markForCheck();
+    this.sdk.currentKb
+      .pipe(
+        take(1),
+        switchMap((kb) =>
+          forkJoin(
+            this.selectedIds.map((id) => {
+              const resource = new Resource(
```

Comment:
The method `getResourceFromData()` can do that for you.
But no need to change, that's fine.